torch/_decomp/decompositions.py

```diff
@@ -1124,7 +1124,10 @@ def _softmax(x: Tensor, dim: int, half_to_float: bool):
         x, type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
     )
     x = x.to(computation_dtype)
-    if x.numel() == 0:
+
+    from torch.fx.experimental.symbolic_shapes import guard_size_oblivious
+
+    if guard_size_oblivious(x.numel() == 0):
```

Comment:
oy oy, I don't think you can do this. The problem is that amax is not defined on zero numel input. 

```
>>> torch.amax(torch.empty(0), dim=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: amax(): Expected reduction dim 0 to have non-zero size.
```

If you know that the input x is non-zero size, you should torch._check that directly in your model code

Reply:
If it can be zero, then we need to do something more complicated here, maybe a use of cond HOP is necessary.

Reply:
Thanks, Ed. Let me check if my understanding is correct - when ```x.numel()``` returns 0, ```guard_size_oblivious(x.numel() == 0)``` will return ```False```, because ```guard_size_oblivious``` assumes ```x``` would not be empty. Consequently, we would fall into the else branch in such a case.  

Let me check if I could find the relevant mode code where we could inject torch._check. Thanks.

Reply:
Yes.